File: DataBinding/Section/XML/C#/XML/StartForm.cs
using System;
using System.Windows.Forms;
using System.Xml;

namespace GrapeCity.ActiveReports.Samples.XML
{
	/// <summary>
	/// Demonstrates how to setup an XML data source through the customer order list.
	/// It also displays subreports and multi-level reports.
	/// </summary>
	public partial class StartForm : System.Windows.Forms.Form
	{
		
		public StartForm()
		{
			// Required for Windows Form Designer support
			//
			InitializeComponent();
		}
	  
		/// <summary>
		/// The main entry point for the application.
		/// </summary>
		[STAThread]
		static void Main() 
		{
			Application.Run(new StartForm());
		}

		/// <summary>
		/// btnCustomers_Click - Checks the radio options and sets the report object's data
		/// </summary>
		private void btnCustomers_Click(object sender, EventArgs e)
		{
			try
			{
				var rpt = new SectionReport();
				rpt.LoadLayout(XmlReader.Create(Properties.Resources.CustomersOrders));

				Data.XMLDataSource ds = rpt.DataSource as Data.XMLDataSource;
				if (ds == null)
				{
					// Display the message when an error occurs.
					MessageBox.Show(Properties.Resources.DataSourceError, this.Text);
					return;
				}

				ds.FileURL = Properties.Resources.ConnectionString;

				// Set an XSL pattern to get the node (record) based on the report.
				if (radioAll.Checked)
				{
					// Show all data
					ds.RecordsetPattern = "//CUSTOMER";
				}
				else if (radioID.Checked)
				{
					// Show data where ID = 2301
					ds.RecordsetPattern = "//CUSTOMER[@id = " + @"""" + "2301" + @"""" + "]";
				}
				else if (radioEmail.Checked)
				{
					// Show data where Email is valid
					ds.RecordsetPattern = "//CUSTOMER[@email]";
				}
				
				ViewerForm frm = new ViewerForm();
				frm.Show();
				frm.LoadReport(rpt);
			}
			catch (ReportException ex)
			{
				MessageBox.Show(ex.ToString(), Text);
			}
		}

		/// <summary>
		/// btnCustomersLeveled_Click - Creates a OrdersLeveled report and sets
		/// the data source for it.
		/// </summary>
		private void btnCustomersLeveled_Click(object sender, EventArgs e)
		{
			try
			{
				//OrdersLeveled rpt = new OrdersLeveled();
				var rpt = new SectionReport();
				rpt.LoadLayout(XmlReader.Create(Properties.Resources.OrdersLeveled));

				Data.XMLDataSource ds = rpt.DataSource as Data.XMLDataSource;
				if (ds == null)
				{
					// Display the message when an error occurs.
					MessageBox.Show(Properties.Resources.DataSourceError, this.Text);
					return;
				}

				// Set the XML file name.
				ds.FileURL = Properties.Resources.ConnectionString;
				
				// Display the report.
				ViewerForm frm = new ViewerForm();
				frm.Show();
				frm.LoadReport(rpt);
			}
			catch (ReportException ex)
			{
				MessageBox.Show(ex.ToString(), Text);
			}
		}
	}
}

File: DataBinding/Section/XML/VB.NET/XML/StartForm.vb
Imports System
Imports System.Windows.Forms
Imports System.Xml
Imports GrapeCity.ActiveReports.SectionReportModel

'Demonstrates how to setup an XML data source through the customer order list.
' It also displays subreports and multi-level reports.
Public Class StartForm
	Inherits System.Windows.Forms.Form

	Public Sub New()
		MyBase.New()

		' Required designer variable.
		InitializeComponent()
	End Sub

	'Clean up any resources being used.
	Protected Overloads Overrides Sub Dispose(ByVal disposing As Boolean)
		If disposing Then
			If Not (components Is Nothing) Then
				components.Dispose()
			End If
		End If
		MyBase.Dispose(disposing)
	End Sub

	'The main entry point for the application.
	<STAThread()>
	Shared Sub Main()
		Application.Run(New StartForm())
	End Sub 'Main

	'btnCustomers_Click - Checks the radio options and sets the report object's data
	Private Sub btnCustomers_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCustomers.Click
		Try
			'Dim rpt As New CustomersOrders()
			Dim rpt As New SectionReport()
			rpt.LoadLayout(XmlReader.Create(My.Resources.CustomersOrders))

			CType(rpt.DataSource, Data.XMLDataSource).FileURL = My.Resources.ConnectionString
			If radioAll.Checked Then 'Show all data
				CType(rpt.DataSource, Data.XMLDataSource).RecordsetPattern = "//CUSTOMER"
			ElseIf radioID.Checked Then 'Show data where ID = 2301
				CType(rpt.DataSource, Data.XMLDataSource).RecordsetPattern = "//CUSTOMER[@id = " + """" + "2301" + """" + "]"
			ElseIf radioEmail.Checked Then 'Show data where Email is valid
				CType(rpt.DataSource, Data.XMLDataSource).RecordsetPattern = "//CUSTOMER[@email]"
			End If

			Dim frm As New ViewerForm()
			frm.Show()
			frm.LoadReport(rpt)
		Catch ex As ReportException
			MessageBox.Show(ex.ToString(), Text)
		End Try
	End Sub 'btnCustomers_Click

	'btnCustomersLeveled_Click - Creates a OrdersLeveled report and sets
	' the data source for it.
	Private Sub btnCustomersLeveled_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCustomersLeveled.Click
		Dim rpt As New SectionReport()
		rpt.LoadLayout(XmlReader.Create(My.Resources.OrdersLeveled))
		CType(rpt.DataSource, Data.XMLDataSource).FileURL = My.Resources.ConnectionString
		rpt.Run()
		Dim frm As New ViewerForm()
		frm.Show()
		frm.LoadReport(rpt)
	End Sub 'btnCustomersLeveled_Click

End Class 'StartForm
